fastlane_core/lib/fastlane_core/device_manager.rb

```diff
@@ -126,18 +169,38 @@ def reset
     end
   end
 
-  class SimulatorTV < Simulator
+  class Simulator < DeviceManager
+    class << self
+      def all
+        return DeviceManager.simulators('iOS')
+      end
+
+      # Reset all simulators of this type
+      def reset_all
+        all.each(&:reset)
+      end
+
+      # Reset simulator by UDID or name and OS version
+      # Latter is useful when combined with -destination option of xcodebuild
+      def reset(udid: nil, name: nil, os_version: nil)
+        match = all.detect { |device| device.udid == udid || device.name == name && device.os_version == os_version }
+        match.reset if match
+      end
+    end
+  end
+
+  class SimulatorTV < DeviceManager
     class << self
-      def requested_os_type
-        'tvOS'
+      def all
+        return DeviceManager.simulators('tvOS')
       end
     end
   end
 
-  class SimulatorWatch < Simulator
+  class SimulatorWatch < DeviceManager
```

Comment:
Good point, i think this had been due to prior experimentation when developing this but not removed when cleaning up. Will amend my commit and separate the classes.